browser/misc_metrics/extension_metrics_factory.h

```diff
@@ -0,0 +1,45 @@
+/* Copyright (c) 2023 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_BROWSER_MISC_METRICS_EXTENSION_METRICS_FACTORY_H_
+#define BRAVE_BROWSER_MISC_METRICS_EXTENSION_METRICS_FACTORY_H_
+
+#include "components/keyed_service/content/browser_context_keyed_service_factory.h"
+#include "components/keyed_service/core/keyed_service.h"
+#include "content/public/browser/browser_context.h"
+
+namespace base {
+template <typename T>
+class NoDestructor;
+}  // namespace base
+
+namespace misc_metrics {
+
+class ExtensionMetrics;
+
+class ExtensionMetricsFactory : public BrowserContextKeyedServiceFactory {
```

Comment:
imo writing an entire service with full set of observers is kinda overengineering for this task (not many people frequently install/unisntall extensions), but let it be :D